content/en/developers/integrations/oauth_for_integrations.md

```diff
@@ -124,22 +153,88 @@ You cannot edit a published OAuth client directly, so only go through the publis
 
 For more information about publishing your integration tile and creating your pull request, see the [Marketplace and Integrations documentation][7].
 
-## Further Reading
+## Troubleshooting
+
+### The list of API scopes does not include sending metrics, events, and logs
+
+To send data to Datadog, use the `api_keys_write` scope when generating an API key on behalf of the user. For more information, see [Create an API key](#create-an-api-key).
+
+### API key and token expiration
+
+Refresh tokens do not expire unless the user revokes authorization or the partner revokes the token. If the partner revokes the token, the user must reauthorize the integration to generate new refresh and access tokens. For more information, see the [OAuth2 Authorization Endpoints Reference][13].
+
+### Retrieving API keys
+
+After you create a key using the [api_keys/marketplace][14] endpoint, the key is returned in the response. The key cannot be regenerated or viewed again. Ensure you store the key securely for continuous data transmission. If you lose your API key, follow these steps to revoke and recreate it:
+
+1. Navigate to the [Datadog API Keys Management page][15].
+1. Look for the API key named `OAuth Client API Key` and select it.
+1. Click **Revoke** to disable the API key.
+1. Follow the steps in [Create an API key](#create-an-api-key) to create a new key.
+1. Reinstall the integration and repeat the OAuth flow.
 
-Additional helpful documentation, links, and articles:
+### Invalid client ID
 
-- [OAuth 2.0 in Datadog][1]
-- [Authorize your Datadog integrations with OAuth][11]
+Error
+: `invalid_request - Invalid client_id parameter value`
 
-[1]: https://docs.datadoghq.com/developers/authorization/oauth2_in_datadog/
+Until an OAuth client is published, you can only authorize the client from the account it was created in (the partner's sandbox account). This error occurs if you try to authorize the client outside of that account before the client is published.
+
+### Forbidden errors related to app keys
+
+OAuth clients use an `access_token` for authentication. Use the `access_token` to make calls to Datadog API endpoints by sending it as a part of the authorization header of your request:
+
+```python
+headers = {"Authorization": "Bearer {}".format(access_token)}
+```
+
+For more information, see [Implement the OAuth protocol][17].
+
+### Forbidden errors on API requests
+
+If you're getting a forbidden error when trying to make an API call to a specific endpoint and you've enabled the correct scope for that endpoint, it's possible your API key, session, or OAuth token is invalid or has expired.
+
+### API errors when making a call with subdomains
+
+When connecting to the Datadog API, do not include the subdomain in the API call. For example, use `datadoghq.eu` instead of `bigcorp.datadoghq.eu`.
```

Comment:
@emarsha94 I went back through our emails and found this (subject line: [Datadog] Update: Re: OAuth2 beta): 'We have a customer, LayerX, who is based in Japan and is using Datadog with a custom subdomain: [layerx-corp-prd.datadoghq.com](http://layerx-corp-prd.datadoghq.com/). They managed to install the Datadog integration into their workspace, which set up a webhook subscription that tries to send events to a URL with their subdomain: https://http-intake.logs.layerx-corp-prd.datadoghq.com/api/v2/logs?ddsource=notion.

However, whenever we try to send an event to the URL we get an SSL cert error:
Hostname/IP does not match certificate's altnames: Host: [http-intake.logs.layerx-corp-prd.datadoghq.com](http://http-intake.logs.layerx-corp-prd.datadoghq.com/). is not in the cert's altnames: DNS:*.[datadoghq.com](http://datadoghq.com/), DNS:[datadoghq.com](http://datadoghq.com/)'. 

I believe this is where this FAQ came from. Can we repurpose the error from that email here to show an example?

Reply:
I don't think this is always the error which is why I don't think we should put one here. For example, the SSO cases where they are using the sub-domain instead of the site.

Reply:
This is perhaps something we can iterate on if we find frequent patterns that lead back to this issue

Reply:
Added the error to this one